src/main/java/org/jenkinsci/plugins/docker/workflow/FromFingerprintStep.java

```diff
@@ -43,6 +43,10 @@
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.DataBoundSetter;
 
+/**
+ * @deprecated Per discussion here: https://github.com/jenkinsci/docker-workflow-plugin/pull/149#issuecomment-451305522
```

Comment:
nit: A link is good, but I would also add a quick inline explanation: 
```suggestion
 * @deprecated Fingerprints produced by this step are not used anywhere, and the parsing code has major limitations. See https://github.com/jenkinsci/docker-workflow-plugin/pull/149#issuecomment-451305522 and https://groups.google.com/d/msg/jenkinsci-dev/k13SfZcBWVg/iQghmCQrEAAJ
```

It looks like this step already override `isAdvanced` and returns true, so it already shows up as advanced/deprecated in the snippet generator, so that's good.

Reply:
And its `help.html` should discuss the deprecation.

Reply:
In particular, it should mention that if you do happen to be one of the people using the Docker Traceability plugin, you can continue to use this step (to the extent its parser works), but you will need to run it manually rather than relying on DSL methods to run it for you.